code/modules/mob/living/basic/vermin/cockroach/cockroach.dm

```diff
@@ -91,6 +92,27 @@
 	minion_path = null
 	ai_controller = /datum/ai_controller/basic_controller/cockroach/aggro
 
+/mob/living/basic/cockroach/bloodroach
+	name = "bloodroach"
+	desc = "This cockroach has gorged itself on maintenance blood, and has a glistening red sheen to its fatty carapace. Incredibly disgusting."
+	icon_state = "bloodroach"
+	icon_dead = "bloodroach_no_animation"
+	health = 3
+	maxHealth = 3 // Wow!!
+	death_drops = list(/obj/effect/decal/cleanable/blood/gibs/old)
+
+/mob/living/basic/cockroach/bloodroach/death(gibbed)
+	if(HAS_TRAIT(src, TRAIT_BUGKILLER_DEATH))
+		return ..()
+	// explode into a pile of blood if not pest controlled
+	for(var/turf/messy_turf in view(src, 2))
+		new /obj/effect/decal/cleanable/blood(messy_turf)
+		for(var/mob/living/mob_in_turf in messy_turf)
```

Comment:
surely we have a better way of doing this that doesn't make such uniform output

Reply:
idk